codechecker_common/source_code_comment_handler.py

```diff
@@ -120,21 +135,45 @@ def __process_source_line_comment(self, source_line_comment):
                 'message': message,
                 'status': review_status}
 
-    def has_source_line_comments(self, source_file, line):
+    def has_source_line_comments(self, fp, line):
         """
-        Return True if there is any source code comment or False if not.
+        Return True if there is any source code comment or False if not,
+        for a given line.
         """
-        comments = self.get_source_line_comments(source_file, line)
+        try:
+            comments = self.get_source_line_comments(fp, line)
+        except SpellException as ex:
+            # Misspell in the review status comment.
+            LOG.warning(ex)
+            return False
         return len(comments)
 
-    def get_source_line_comments(self, source_file, bug_line):
+    def scan_source_line_comments(self, fp, line_numbers):
+        """collect all the source line review comments if exists
+        in a source file at the given line numbers.
+
+        returns a list of (line_num, comments) tuples where comments
+        were found.
         """
-        This function returns the available preprocessed source code comments
-        for a bug line.
+        comments = []
+        misspelled_comments = []
+        if not contains_codechecker_comment(fp):
+            return comments, misspelled_comments
+
+        line_numbers = sorted(line_numbers)
+        for num in line_numbers:
+            try:
+                comments.append((num, self.get_source_line_comments(fp, num)))
```

Comment:
yes, there might be some further optimizations, but this jumping and going through the lines is done in memory the whole file is read once into memory, so jumping back to the start should not be a problem, iterating over the lines multiple times yes it can take more time, but just iterating and selecting the required lines which is done.
Probably it could be further optimized if this is the bottleneck after doing some measurements.